startup/GafferScene/renderPassTypes.py

```diff
@@ -0,0 +1,520 @@
+##########################################################################
+#
+#  Copyright (c) 2024, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import imath
+import inspect
+
+import IECore
+import IECoreScene
+
+import Gaffer
+import GafferScene
+
+def __renderPassShaderAssignment( usage ):
+
+	processor = GafferScene.SceneProcessor()
+
+	processor["renderer"] = Gaffer.StringPlug()
+
+	processor["defaultShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+	processor["defaultArnoldShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+	processor["defaultCyclesShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
+	processor["defaultDelightShader"] = Gaffer.ObjectPlug( defaultValue = IECore.NullObject() )
```

Comment:
Hardcoding renderer-specific stuff in here seems unfortunate - ideally we'd want people to be able to add new renderers completely non-intrusively. I think maybe the way we'd want to do that would be to replace the `default{renderer}Shader` inputs with fallbacks to the default value for the `renderPass:shader:{usage}:{renderer}` option, so that anyone can register a default from anywhere via the Metadata API?

I know we haven't fully resolved the question of if we want to use metadata within computes (and thus mandate that the metadata must be static after startup), so maybe the pragmatic thing to do now is just add a `\todo` that we don't want to hardcode this stuff in the long run?

Reply:
Yeah, it didn't feel great while setting that up, I've added the `\todo` for now in 057f8553ef08d8e9d20d659210b09a57ad4d20d6.